packages/grid/x-data-grid/src/hooks/features/export/useGridPrintExport.tsx

```diff
@@ -71,12 +71,6 @@ export const useGridPrintExport = (
   const updateGridColumnsForPrint = React.useCallback(
     (fields?: string[], allColumns?: boolean) =>
       new Promise<void>((resolve) => {
-        // TODO remove unused Promise
-        if (!fields && !allColumns) {
```

Comment:
The subsequent logic handles `disableExport` flag, I didn't duplicate only the `disableExport` handling in this `if` block because even if the `!fields && !allColumns` computes to `true`, the cost for subsequent logic is negligible, and it anyways tests for them being truthy or not.